packages/codd/scripts/codd.py

```
@@ -175,6 +304,95 @@ def commit_time(root: Path, rel_path: str) -> float:
         return 0.0  # 取得不能なら最古扱い（drift の誤検知を防ぐ）
 
 
+def _dirty_paths(root: Path) -> set[str] | None:
+    """``git status --porcelain -z`` からリポジトリ全体の dirty パス集合を返す。
+
+    未追跡・未コミット編集のパス（rename の旧パス・新パス両方を含む）を 1 回の
```

Ruff (0.15.21)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)